thoth-api/src/graphql/model.rs

```diff
@@ -2436,18 +2483,74 @@ impl Work {
     }
 
     #[graphql(description = "Concatenation of title and subtitle with punctuation mark")]
-    pub fn full_title(&self) -> &str {
-        self.full_title.as_str()
+    pub fn full_title(&self, ctx: &Context) -> FieldResult<String> {
+        let mut connection = ctx.db.get()?;
+        let title = title::table
+            .filter(title::work_id.eq(&self.work_id))
+            .filter(title::canonical.eq(true))
+            .first::<Title>(&mut connection)
+            .map_err(|e| FieldError::new(e.to_string(), Value::null()))?;
```

Comment:
There's no need to inisialise `FieldError`, you can just `.map_err(Into::into)`

Reply:
In fact, you don't need the `map_err` at all, you can just `.first::<Title>(&mut connection)?;`